src/Extension/FluentDirectorExtension.php

```diff
@@ -166,9 +166,9 @@ protected function getRuleController($existingRule, $localeObj)
      */
     protected function insertRuleBefore(array $rules, $key, array $rule, $prependIfMissing = true)
     {
-        $i = array_search($key, array_keys($rules));
+        $i = array_search($key, array_keys($rules ?? []));
```

Comment:
I assume `array` param type precludes null values. Is that not true in 8.1 anymore?

Reply:
ok,  so it's more to do with changes to internal function arguments than typehints, apparently. I think in this case we can rely on the typehint. :)